app/src/main/java/com/example/arc_exapmle/activity/LoginActivity2nd.kt

```diff
@@ -0,0 +1,143 @@
+
+    companion object {
+        const val loginValue = "id"
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_login_two)
+
+        loginActivityViewModel = ViewModelProviders.of(
+            this,
+            LoginActivityViewModelFactory(UserRepository(StarDatabase.getInstance(this).userDao()))
+        )
+            .get(LoginActivityViewModel::class.java)
+
+        setType()
+        loginOnClick()
+        inputChange()
+    }
+
+    private fun setType() {
+        idEditText = findViewById(R.id.email)
+        login_button = findViewById(R.id.login_button)
+        login_button_card_view = findViewById(R.id.login_button_card_view)
+    }
+
+    @SuppressLint("ResourceType")
+    private fun loginButtonStyle() {
+        if (idEditText.text.isNotEmpty()) {
+            if (!login_button.isFocusable) {
+                login_button.isFocusable = true
+                login_button.isClickable = true
+                login_button_card_view.setCardBackgroundColor(Color.parseColor(getString(R.color.colorAccent)))
+                val outValue = TypedValue()
+                theme.resolveAttribute(android.R.attr.selectableItemBackground, outValue, true)
+                login_button.setBackgroundResource(outValue.resourceId)
+            }
+        } else {
+            if (login_button.isFocusable) {
+                login_button.isFocusable = false
+                login_button.isClickable = false
+                login_button_card_view.setCardBackgroundColor(Color.parseColor(getString(R.color.colorCardViewBackground)))
+                login_button.setBackgroundResource(0)
+            }
+        }
+    }
+
+    private fun loginOnClick() {
+        login_button.setOnClickListener { view ->
+            if (idEditText.text.isNotEmpty()) {
+
+                loginActivityViewModel.userEntry(idEditText.text.toString())
+
+            }
+        }
+    }
+
+    private fun inputChange() {
+        idEditText.addTextChangedListener(object : TextWatcher {
+            override fun beforeTextChanged(
+                charSequence: CharSequence,
+                start: Int,
+                count: Int,
+                after: Int
+            ) {
+
+            }
+
+            override fun onTextChanged(
+                charSequence: CharSequence,
+                start: Int,
+                count: Int,
+                after: Int
+            ) {
+                loginButtonStyle()
+            }
+
+            override fun afterTextChanged(editable: Editable) {
+
+            }
+        })
+    }
+
+    override fun onResume() {
+        super.onResume()
+
+        loginActivityViewModel.onSuccessMutableLiveData.observe(this , Observer {
+
+            val mainIntent = Intent(this, MainActivity::class.java)
+
+            mainIntent.putExtra(
+                loginValue,
+                UserUI(it.username, Integer.parseInt(it.userId))
```

Comment:
chera ino to viewModel nemisazi?

Reply:
Done.
but honestly, I don't understand why that matters